src/chttpd/test/eunit/chttpd_changes_test.erl

```diff
@@ -492,6 +507,107 @@ t_view_filter_no_match({_, DbUrl}) ->
     ?assertEqual({8, 0, []}, changes(DbUrl, Params)),
     {200, #{}} = req(delete, DDocUrl ++ "?rev=" ++ binary_to_list(Rev)).
 
+t_include_docs({_, DbUrl}) ->
+    Params = "?include_docs=true",
+    {200, #{<<"results">> := Res1}} = req(get, DbUrl ++ "/_changes" ++ Params),
+    ?assertMatch(
+        [
+            #{
+                <<"changes">> := [#{<<"rev">> := <<"2-c">>}],
+                <<"doc">> := #{<<"_id">> := ?DOC1, <<"_rev">> := <<"2-c">>}
+            },
+            #{
+                <<"changes">> := [#{<<"rev">> := <<"2-b">>}],
+                <<"doc">> := #{
+                    <<"_id">> := ?DOC3, <<"_rev">> := <<"2-b">>, <<"_deleted">> := ?DELETED
+                }
+            },
+            #{
+                <<"changes">> := [#{<<"rev">> := <<"2-c">>}],
+                <<"doc">> := #{<<"_id">> := ?DDOC2, <<"_rev">> := <<"2-c">>}
+            }
+        ],
+        Res1
+    ),
+    {200, #{<<"results">> := Res2}} = req(post, DbUrl ++ "/_changes" ++ Params),
+    ?assertEqual(Res1, Res2).
+
+t_all_docs_include_docs({_, DbUrl}) ->
+    Params = "?style=all_docs&include_docs=true",
+    {200, #{<<"results">> := Res1}} = req(get, DbUrl ++ "/_changes" ++ Params),
+    ?assertMatch(
+        [
+            #{
+                <<"changes">> := [#{<<"rev">> := <<"2-c">>}, #{<<"rev">> := <<"2-b">>}],
+                <<"doc">> := #{<<"_id">> := ?DOC1, <<"_rev">> := <<"2-c">>}
+            },
+            #{
+                <<"changes">> := [#{<<"rev">> := <<"2-b">>}],
+                <<"doc">> := #{
+                    <<"_id">> := ?DOC3, <<"_rev">> := <<"2-b">>, <<"_deleted">> := ?DELETED
+                }
+            },
+            #{
+                <<"changes">> := [#{<<"rev">> := <<"2-c">>}],
+                <<"doc">> := #{<<"_id">> := ?DDOC2, <<"_rev">> := <<"2-c">>}
+            }
+        ],
+        Res1
+    ),
+    {200, #{<<"results">> := Res2}} = req(post, DbUrl ++ "/_changes" ++ Params),
+    ?assertEqual(Res1, Res2).
+
+t_selector_include_docs({_, DbUrl}) ->
+    Body = #{<<"selector">> => #{<<"_id">> => ?DOC1}},
+    Params = "?filter=_selector&include_docs=true",
+    {200, #{<<"results">> := Res}} = req(post, DbUrl ++ "/_changes" ++ Params, Body),
+    ?assertMatch(
+        [
+            #{
+                <<"changes">> := [#{<<"rev">> := <<"2-c">>}],
+                <<"doc">> := #{<<"_id">> := ?DOC1, <<"_rev">> := <<"2-c">>}
+            }
+        ],
+        Res
+    ).
+
+t_selector_all_docs_include_docs({_, DbUrl}) ->
+    Body = #{<<"selector">> => #{<<"_id">> => ?DOC1}},
+    Params = "?filter=_selector&style=all_docs&include_docs=true",
+    {200, #{<<"results">> := Res}} = req(post, DbUrl ++ "/_changes" ++ Params, Body),
+    ?assertMatch(
+        [
+            #{
+                <<"changes">> := [#{<<"rev">> := <<"2-c">>}, #{<<"rev">> := <<"2-b">>}],
+                <<"doc">> := #{<<"_id">> := ?DOC1, <<"_rev">> := <<"2-c">>}
+            }
+        ],
+        Res
+    ).
```

Comment:
For completeness, let's add two more test which also sets `conflicts=true` parameter, one with `style=all_docs` and one with default style. There we would be checking that the document body returned has `_conflicts: ...` fields as expected.

Reply:
To include the `_conflicts` field, it seems I still need to open the doc one more time [here](https://github.com/apache/couchdb/pull/4862/files#diff-e8d45123acc0305191b7564c24d8a56e76b8d50e6c6ca66da3ae5c5d13c20d6bR388-R389).
If this version is acceptable, I'll add more test cases for `view / custom`  etc.

Reply:
Oh interesting. Is that because the document didn't have the `_conflicts` field otherwise.

Reply:
How about this time? Added special case for `filter/5` when `style=all_docs&conflicts=true&include_docs=true` and `open_revs2()` to modify the response to include the `_conflicts` field.